modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -949,28 +1004,64 @@ private Page<Account> _getOrganizationAccountsPage(
 					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
 	}
 
-	private long[] _getOrganizationIds(Account account) {
+	private long[] _getOrganizationIds(Account account) throws Exception {
 		Long[] organizationIds = account.getOrganizationIds();
 
-		if (organizationIds == null) {
-			return new long[0];
+		if (ArrayUtil.isEmpty(organizationIds)) {
+			String[] organizationExternalReferenceCodes =
+				account.getOrganizationExternalReferenceCodes();
+
+			if (organizationExternalReferenceCodes == null) {
+				return new long[0];
+			}
+
+			organizationIds =
+				new Long[organizationExternalReferenceCodes.length];
+			int index = 0;
+
+			for (String externalReferenceCode :
+					organizationExternalReferenceCodes) {
+
+				Organization organization = _toOrganization(
+					_organizationService.
+						fetchOrganizationByExternalReferenceCode(
+							externalReferenceCode,
+							contextCompany.getCompanyId()));
+
+				if (organization != null) {
+					organizationIds[index++] = Long.valueOf(
+						organization.getId());
+				}
+			}
```

Comment:
doesn't exist something to "do this"? Something like a transform util or a ListUtils that takes an array and call a custom function for each element?